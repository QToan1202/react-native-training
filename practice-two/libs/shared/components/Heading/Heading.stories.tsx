import { ComponentMeta, ComponentStory } from '@storybook/react-native'

import { Heading } from '../index'

export default {
  title: 'components/Heading',
  component: Heading,
} as ComponentMeta<typeof Heading>

const Template: ComponentStory<typeof Heading> = (args) => <Heading {...args} />

export const Default = Template.bind({})
Default.args = {
  content: 'My default heading',
}

export const MediumHeading = Template.bind({})
MediumHeading.args = {
  content: 'My medium heading',
  fontWeight: '$2',
}

export const HeadingWithCustomStyle = Template.bind({})
HeadingWithCustomStyle.args = {
  content: 'My custom heading',
  fontWeight: '$2',
  color: '#000',
}
