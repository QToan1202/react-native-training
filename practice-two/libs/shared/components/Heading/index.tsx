import { memo } from 'react'
import isEqual from 'react-fast-compare'

import StyledHeading, { StyledHeadingProps } from './styles'

export type HeadingProps = {
  content: string
} & StyledHeadingProps

const Heading = ({ content, ...rest }: HeadingProps) => {
  return (
    <StyledHeading size="$5" fontWeight="$4" color="$color.white" letterSpacing="$2" {...rest}>
      {content}
    </StyledHeading>
  )
}

export default memo(Heading, isEqual)
