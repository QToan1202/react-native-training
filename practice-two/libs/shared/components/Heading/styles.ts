import { GetProps, Heading, styled } from 'tamagui'

const StyledHeading = styled(Heading, {
  variants: {
    size: {
      '...fontSize': (value, { font }) => {
        return {
          fontSize: font?.size[value] ?? value,
          lineHeight: font?.lineHeight[value] ?? value,
        }
      },
    },
  } as const,
})

export type StyledHeadingProps = GetProps<typeof StyledHeading>

export default StyledHeading
