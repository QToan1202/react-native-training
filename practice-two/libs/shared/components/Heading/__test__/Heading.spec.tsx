import { render } from '@testing-library/react-native'
import Heading from '../index'

test('Render heading component', () => {
  const component = render(<Heading content="my heading" textMedium />).toJSON()

  expect(component).toMatchSnapshot()
})
