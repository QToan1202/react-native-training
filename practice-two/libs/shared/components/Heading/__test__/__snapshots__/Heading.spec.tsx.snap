// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render heading component 1`] = `
<Text
  style={
    [
      {
        "color": "#fff",
        "fontFamily": "Montserrat_700Bold",
        "fontSize": 24,
      },
      {
        "fontFamily": "Montserrat_500Medium",
      },
      undefined,
    ]
  }
>
  my heading
</Text>
`;
